mock/py/mockbuild/util.py

```diff
@@ -903,12 +897,15 @@ def _prepare_nspawn_command(chrootPath, user, cmd, nspawn_args=None, env=None,
     if env:
         for k, v in env.items():
             nspawn_argv.append('--setenv={0}={1}'.format(k, v))
-    cmd = nspawn_argv + cmd
-    if cmd_is_list:
-        return cmd
-    else:
-        return " ".join(cmd)
 
+    # /bin/sh -c wrapper explicitly requested
+    if shell and isinstance(cmd, list):
+        cmd = ' '.join(cmd)
+
+    if isinstance(cmd, str):
+        cmd = ['/bin/sh', '-c', cmd]
```

Comment:
??? this is not based on `shell` variable?

Reply:
Yes, that's the hack needed for `--chroot 'one command'`, which needs to stay expanded.
And we always drop the `shell=True` after calling _prepare_nspawn ... so :-( this is 
not something I'm happy about either.

Reply:
I'll at least document the reasons here.